RadarSDK/RadarUser.m

```diff
@@ -184,11 +189,41 @@ - (instancetype _Nullable)initWithObject:(NSObject *)object {
             }
         }
         
-        userPlaceChains = mutableNearbyChains;
+        userNearbyPlaceChains = mutableNearbyChains;
+    }
+    
+    id userSegmentsObj = userDict[@"segments"];
+    if ([userSegmentsObj isKindOfClass:[NSArray class]]) {
+        NSArray *segmentsArr = (NSArray *)userSegmentsObj;
+        
+        NSMutableArray<RadarSegment *> *mutableSegments = [NSMutableArray<RadarSegment *> new];
+        for (id segmentObj in segmentsArr) {
+            RadarSegment *segment = [[RadarSegment alloc] initWithObject:segmentObj];
+            if (segment) {
+                [mutableSegments addObject:segment];
+            }
+        }
+        
+        userSegments = mutableSegments;
+    }
+    
+    id userTopChainsObj = userDict[@"topChains"];
```

Comment:
for consistency, can we put the array deserializers in the class -- https://github.com/radarlabs/radar-sdk-android/blob/c661f33d6fd9482c9e47c57a1746ec8cd27f2649/sdk/src/main/java/io/radar/sdk/model/RadarChain.kt#L50-L58

Reply:
I'm going to come back and do a final cleanup pass on the deserializers and the test API responses, but will do that in a separate PR